modules/user-interface-configuration/src/main/java/org/opencastproject/uiconfig/UIConfigRest.java

```diff
@@ -167,10 +167,11 @@ public Response getConfigFile(@PathParam("component") String component, @PathPar
 
     try {
       final String basePath = new File(uiConfigFolder, orgId).getCanonicalPath();
+      final String basePathWithSep = basePath.endsWith(File.separator) ? basePath : basePath + File.separator;
       final String configFileCanPath = configFile.getCanonicalPath();
 
       // is configFile a subdirectory of basePath (additional directory traversal protection), if not stop
-      if (!configFileCanPath.startsWith(basePath)) {
+      if (!configFileCanPath.startsWith(basePathWithSep)) {
```

Comment:
In combination with the change above:
```suggestion
      if (!configFileCanPath.startsWith(basePath)) {
```